### PR TITLE
Bug 1208347 - Add MozIntl API

### PR DESCRIPTION
**Pull request by zbraniecki**
https://bugzilla.mozilla.org/show_bug.cgi?id=1208347

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=9b283070cd40fac10baab38a3737fa810ad469b7

File: shared/js/moz_intl.js [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+   *   month: 'long'
+   * });
+   * 
+   * f.format(date, {
+   *  day: '<strong>$&</strong>',
+   * });
+   *
+   * Warning: Current implementation is very fragile. Be very careful when
+   * choosing tokens to be formatted. The formatter will replace the
+   * first occurance of the returned string, so if you try to format
```
Comment (stasm) [inline shared/js/moz_intl.js]: Nit: typo in 'occurrence'.

File: apps/sharedtest/test/unit/moz_intl_test.js [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+    navigator.mozL10n = MockL10n;
+  });
+
+  suiteTeardown(function() {
+    navigator.mozL10n = realL10n;
+  });
+
+  suite('listFormatter', function() {
+    setup(function() {
+      sinon.stub(navigator.mozL10n, 'formatValue', function(...args) {
```
Comment (stasm) [inline apps/sharedtest/test/unit/moz_intl_test.js]: I think you might be able to use `this.sinon` here which is a Sinon sandbox created automatically for each test suite: https://github.com/mozilla-b2g/gaia/blob/2bdae877bc22c2abbb080c29f390a4d4b961d677/dev_apps/test-agent/common/test/sinon_helper.js#L4.  You wouldn't need to call `restore` in `teardown`.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=fb7cd30b902bf7c0c8a198a40afc0a39df767a24

**Pull request merged**